jobs/openshift/cluster/brew/Jenkinsfile

```diff
@@ -0,0 +1,143 @@
+    imageSyncOptions.clusterAdminPassword = params.clusterAdminPassword ?: clusterAdminCredentials.clusterAdminPassword // Defaults to the password stored in tower-openshift-cluster-credentials
+    imageSyncOptions.installationGitUrl = params.installationGitUrl
+    imageSyncOptions.installationGitBranch = params.installationGitBranch
+    verifyImageSyncOptions(imageSyncOptions)
+    
+    println "Image Sync Options: ${imageSyncOptions}"
+    currentBuild.displayName = "${currentBuild.displayName} ${imageSyncOptions.installationGitBranch}"
+    currentBuild.description = imageSyncOptions.openshiftMasterUrl
+}
+
+def githubSSHCredentialsID = 'jenkinsgithub'
+def imageSyncConfigFile = './integreatly.yml'
+String brewContainerRegistry
+def projectsToSync
+def syncImages = false
+
+node('cirhos_rhel7') {
+    cleanWs()
+    stage('Ensure Sync File Exists') {
+        dir('installation') {
+            gitCheckoutRepo(imageSyncOptions.installationGitUrl, imageSyncOptions.installationGitBranch, githubSSHCredentialsID, '.')
+            syncImages = fileExists imageSyncConfigFile
+
+            if (!syncImages) {
+                println "Image sync file '${imageSyncConfigFile}' is not available. No images to sync, skipping!"
+            }
+        }
+    }
+
+    stage('Get Images To Sync') {
+        when(syncImages) {
+            dir('installation') {
+                def imageSyncConfigContent = readYaml file: imageSyncConfigFile
+                projectsToSync = imageSyncConfigContent.imageSync
+                brewContainerRegistry = imageSyncConfigContent.sourceRepo
+            }
+        }
+    }
+
+    stage('Authenticate to OpenShift') {
+        when(syncImages) {
+            sh "oc login ${imageSyncOptions.openshiftMasterUrl} -u ${imageSyncOptions.clusterAdminUsername} -p ${imageSyncOptions.clusterAdminPassword} --insecure-skip-tls-verify"
+            imageSyncOptions.clusterAdminAuthToken = sh(returnStdout: true, script: "oc whoami -t").trim()
+        }
+    }
+
+    stage('Create Brew Projects') {
+        when(syncImages) {
+            String clusterContainerRegistryConsole = "registry-console-default.apps.${getHost(imageSyncOptions.openshiftMasterUrl)}" 
+            projectsToSync.keySet().each {
+                if (params.dryRun) {
+                    println "Would create a new project called '${it}' and set the registry project access policy to 'Anonymous: Allow all unauthenticated users to pull images'"
+                } else {
+                    sh "oc new-project ${it} || true"
+                    // Set registry project access policy to Anonymous: Allow all unauthenticated users to pull images
+                    sh "oc create rolebinding registry-viewer --clusterrole=registry-viewer || true"
+                    sh "oc adm policy add-role-to-group registry-viewer system:authenticated system:unauthenticated || true"
```

Comment:
Cool, so this will allow any user/project to pull images from any of these registries we are syncing right?

Reply:
yes this should allow that. If you go into the registry project, it should show that the access policy is set to `Anonymous: Allow all unauthenticated users to pull images`